openedx/frontend-app-course-authoring
# feat: [FC-0044] Course unit page - Unit switch widget with a New unit creation button

### Description

Unit switch widget with a New unit creation button for the Course unit page. This functionality has been migrated and adapted from [MFE Learning](https://github.com/openedx/frontend-app-learning).

The primary features were implemented:
- Course unit switch widget: the transfer and partial adaptation of the display and navigation functionality units of the [MFE Learning](https://github.com/openedx/frontend-app-learning) was carried out. The CMS will be used later instead of LMS.
- New unit button: added display of the New unit button (at this stage, it is disabled).

### Developer notes
- The initial commit in this pull request is temporary. Once the https://github.com/openedx/frontend-app-course-authoring/pull/808 is merged, the second commit of this pull request will become the main commit.
- The code base for this pull request does not provide test coverage.
- The current implementation of the unit switch widget provides the essential functionality required for the subsequent development of the Course unit page.
- In the upcoming pull requests, the LMS endpoints of the MFE Learning will be substituted with distinct and independent endpoints from the CMS.

### Design
https://www.figma.com/file/YeKFwSpyLaJFDs3f3p3TSa/Studio-1%3A1-mock-ups?type=design&node-id=599-23595&mode=design&t=icsTrD5n70NSEnND-0
<img width="1105" alt="image" src="https://github.com/openedx/frontend-app-course-authoring/assets/93188219/f04ae097-c860-4ae8-bad2-6ce8d71b09ec">

### Testing instructions
1. Run master devstack.
2. Start platform make dev.up.lms+cms+frontend-app-course-authoring and make checkout on this branch.
3. Enable the new Unit page by adding a waffle flag `contentstore.new_studio_mfe.use_new_unit_page` in the CMS admin panel.
4. Make sure that the MFE setting `ENABLE_UNIT_PAGE=true` is enabled.
5. Go to the Course Unit page from the Course Outline page.
6. Make sure the course you are viewing is not outdated.
7. Publish all sections on the Course Outline page.
